operator/cmd/root.go

```diff
@@ -111,25 +116,35 @@ func Execute() {
 	}
 }
 
-func registerOperatorHooks(lc hive.Lifecycle, clientset k8sClient.Clientset, shutdowner hive.Shutdowner) {
+func registerOperatorHooks(lc hive.Lifecycle, llc *LeaderLifecycle, clientset k8sClient.Clientset, shutdowner hive.Shutdowner) {
 	k8s.SetClients(clientset, clientset.Slim(), clientset, clientset)
 	initEnv()
 
 	lc.Append(hive.Hook{
 		OnStart: func(context.Context) error {
-			go runOperator(clientset, shutdowner)
+			go runOperator(llc, clientset, shutdowner)
 			return nil
 		},
-		OnStop: func(context.Context) error {
+		OnStop: func(ctx context.Context) error {
+			if err := llc.Stop(ctx); err != nil {
+				return err
+			}
 			doCleanup()
 			return nil
 		},
 	})
 }
 
+// LeaderLifecycle wraps hive.DefaultLifecycle to give it a unique type name.
```

Comment:
Should we separate this into its own commit? Happy to also try to drive it in through a PR of its own.